terraform/modules/psn/main.tf

```diff
@@ -82,3 +82,18 @@ resource "aws_route53_record" "psn_regioned_service" {
   records = [aws_vpc_endpoint.psn_service.dns_entry[count.index + 1]["dns_name"]]
 }
 
+data "aws_network_interface" "psn_interfaces" {
+  count = length(aws_vpc_endpoint.psn_service.network_interface_ids)
+
+  id = aws_vpc_endpoint.psn_service.network_interface_ids[count.index]
+}
+
+data "template_file" "psn_network_policy_yaml" {
+  template = file("${path.module}/data/psn-network-policy.yaml")
+
+  psn_cidrs = jsonencode(formatlist("%s/32", data.aws_network_interface.psn_interfaces[*].private_ip))
```

Comment:
can we not directly use aws_vpc_endpoint.psn_service.network_interface_ids[*].private_ip without the extra psn_interfaces local data thing?

Reply:
also, shouldn't this be inside a `vars = { ... }` block?

Reply:
> can we not directly use aws_vpc_endpoint.psn_service.network_interface_ids[*].private_ip without the extra psn_interfaces local data thing?

I don't think so. The `network_interface_ids` attribute is just a list of strings, rather than a list of objects describing each network interface. At least, that's the impression I get from the docs.

> shouldn't this be inside a `vars = { ... }` block?

Oof. I've got rid of it in favour of `templatefile` as you suggest.